.ci/Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent {label "clx-host-21514||clx-host-21518"}
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'10'))
+        timeout(time: 3, unit: 'MINUTES')
+        disableConcurrentBuilds()
+    }
+
+    environment {
+        SRC_ROOT = "${env.WORKSPACE}"
+        ARTEFACT_DIR = "${env.WORKSPACE}/artefacts"
+        NFS_WORKSPACE = "/.autodirect/mtrswgwork/_ml_devops_team/workspace/ml-nccl-rdma-sharp-plugins-pr/${env.BUILD_NUMBER}"
+        CUDA_HOME = "/hpc/local/oss/cuda10.0"
+        NCCL_DIR = "/.autodirect/mtrswgwork/artemry/ci_tools_do_not_remove/nccl-sharp-stable-2019-08-16/build"
+        NCCL_TESTS_DIR = "${NFS_WORKSPACE}/nccl-tests"
+        // Variables for publish_artefacts_to_gist.py
+        UPSTREAM_JOB_NAME = "${env.JOB_NAME}"
+        UPSTREAM_BUILD_NUMBER = "${env.BUILD_NUMBER}"
+        UPSTREAM_ghprbGhRepository = "${env.ghprbGhRepository}"
+        UPSTREAM_ghprbPullId = "${env.ghprbPullId}"
+    }
+
+    stages {
+        stage('Preparations') {
+            steps {
+                echo 'Preparations...'
+                sh 'mkdir -p ${ARTEFACT_DIR}'
+                sh 'mkdir -p ${NFS_WORKSPACE}'
+            }
+        }
+        stage('Checkout NCCL tests') {
+            steps {
+                sh 'mkdir -p ${NCCL_TESTS_DIR}'
+                dir("${NCCL_TESTS_DIR}") {
+                    git branch: 'master',
+                    url: 'https://github.com/NVIDIA/nccl-tests.git'
+                }
+            }
+        }
+        stage('Build') {
+            steps {
+                echo 'Building nccl-rdma-sharp-plugins...'
+                sh 'set -o pipefail; ${SRC_ROOT}/.ci/build.sh 2>&1 | tee ${ARTEFACT_DIR}/build.log'
```

Comment:
IMO - pipefail and other shell knobs should be rurned inside build.sh script

Reply:
This pipefail option is specifically for this one-line pipe - otherwise 'tee' sets its own exit code (typically 0). Inside the shell scripts there may be another pipefail for script itself.